File: Backend/Services/Company/Implementations/CompanyService.cs
using Backend.DTOs.Company;
using Backend.Models.Employer;
using Backend.Repositories.Company.Interfaces;
using Backend.Services.Company.Interfaces;

namespace Backend.Services.Company.Implementations
{
    public class CompanyService : ICompanyService
    {
        private readonly ICompanyRepository _companyRepository;

        public CompanyService(ICompanyRepository companyRepository)
        {
            _companyRepository = companyRepository;
        }

        public async Task<Models.Employer.Company> CreateAsync(
            CreateCompanyDto dto)
        {
            var company = new Models.Employer.Company
            {
                CompanyName = dto.CompanyName,
                Description = dto.Description,
                Industry = dto.Industry,
                Location = dto.Location,
                Website = dto.Website,
                ContactEmail = dto.ContactEmail,
                ContactPhone = dto.ContactPhone
            };

            return await _companyRepository.CreateAsync(company);
        }

        public async Task<Models.Employer.Company?> GetByIdAsync(
    int companyId)
        {
            return await _companyRepository.GetByIdAsync(companyId);
        }

        public async Task<bool> UpdateAsync(
    int companyId,
    UpdateCompanyDto dto)
        {
            var company = await _companyRepository.GetByIdAsync(companyId);

            if (company == null)
            {
                return false;
            }

            company.CompanyName = dto.CompanyName;
            company.Description = dto.Description;
            company.Industry = dto.Industry;
            company.Location = dto.Location;
            company.Website = dto.Website;
            company.ContactEmail = dto.ContactEmail;
            company.ContactPhone = dto.ContactPhone;
            company.UpdatedAt = DateTime.UtcNow;

            return await _companyRepository.UpdateAsync(company);
        }
    }
}

File: Frontend/src/app/core/services/company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, tap } from 'rxjs';
import { AuthService } from './auth.service';

export interface CompanyDto {
  companyName: string;
  description: string | null;
  industry: string | null;
  location: string | null;
  website: string | null;
  contactEmail: string | null;
  contactPhone: string | null;
}

export interface CompanyModel extends CompanyDto {
  companyId: number;
  createdAt: string;
  updatedAt: string | null;
}

@Injectable({
  providedIn: 'root',
})
export class CompanyService {
  private readonly apiUrl = 'https://localhost:7182/api/employer/company';

  constructor(private http: HttpClient, private authService: AuthService) {}

  getCurrentCompanyId(): number | null {
    return this.authService.getCurrentCompanyId();
  }

  createCompany(dto: CompanyDto): Observable<CompanyModel | null> {
    const requestToken = this.authService.getToken();
    return this.http.post<CompanyModel | null>(this.apiUrl, dto).pipe(
      tap(company => {
        if (company && Number.isInteger(company.companyId) && company.companyId > 0) {
          this.authService.setCompanyId(company.companyId, requestToken);
        }
      })
    );
  }

  getCompany(companyId: number): Observable<CompanyModel | null> {
    return this.http.get<CompanyModel | null>(`${this.apiUrl}/${companyId}`);
  }

  updateCompany(companyId: number, dto: CompanyDto): Observable<{ message: string } | null> {
    return this.http.put<{ message: string } | null>(`${this.apiUrl}/${companyId}`, dto);
  }
}
